File: WithTable/app/src/main/java/com/pos_billingwala/Extra/MessPayerMode.java
package com.pos_billingwala.Extra;

import android.content.Context;

/**
 * Mess-level payer mode: user (default) or institute.
 * Cached locally; synced via mess_shop_setting_* APIs.
 */
public final class MessPayerMode {

    public static final String MODE_USER = "user";
    public static final String MODE_INSTITUTE = "institute";
    private static final String PREF_KEY = "mess_payer_mode";

    private MessPayerMode() {
    }

    public static String normalize(String mode) {
        if (mode != null && MODE_INSTITUTE.equalsIgnoreCase(mode.trim())) {
            return MODE_INSTITUTE;
        }
        return MODE_USER;
    }

    public static boolean isInstitutePay(Context context) {
        return MODE_INSTITUTE.equals(get(context));
    }

    public static String get(Context context) {
        if (context == null) {
            return MODE_USER;
        }
        return normalize(Common.getSavedUserData(context, PREF_KEY));
    }

    public static void setLocal(Context context, String mode) {
        if (context == null) {
            return;
        }
        Common.saveUserData(context, PREF_KEY, normalize(mode));
    }
}

File: posbillingwala-flutter/lib/features/mess/domain/mess_payer_mode.dart
import 'package:shared_preferences/shared_preferences.dart';

/// Mess-level payer mode: user (default) or institute.
/// Cached locally; synced via `mess_shop_setting_*` APIs (WithTable `MessPayerMode`).
abstract final class MessPayerMode {
  MessPayerMode._();

  static const String modeUser = 'user';
  static const String modeInstitute = 'institute';
  static const String prefKey = 'mess_payer_mode';

  static String normalize(String? mode) {
    if (mode != null && mode.trim().toLowerCase() == modeInstitute) {
      return modeInstitute;
    }
    return modeUser;
  }

  static Future<bool> isInstitutePay() async {
    return (await get()) == modeInstitute;
  }

  static Future<String> get() async {
    final prefs = await SharedPreferences.getInstance();
    return normalize(prefs.getString(prefKey));
  }

  static Future<void> setLocal(String mode) async {
    final prefs = await SharedPreferences.getInstance();
    await prefs.setString(prefKey, normalize(mode));
  }
}
